### PR TITLE
Feature/2572/poll cloud events

### PR DESCRIPTION
**Pull request by warber**
part of [2572](https://github.com/keptn/keptn/issues/2572)

This PR includes the main logic for watching events

### CODE REVIEW

File: pkg/api/utils/eventwatch.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			continue
+		case <-ew.timeout:
+			close(ch)
+			return
+		case <-ctx.Done():
+			close(ch)
+			return
+		}
+	}
+}
```
Comment (christian-kreuzberger-dtx) [inline pkg/api/utils/eventwatch.go:49]: Please add comments to this code block, telling us what each `case` does and why

File: pkg/api/utils/eventwatch.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+// Watch starts the watch loop and returns a channel to get the actual events as well as a context.CancelFunc in order
+// to stop the watch routine
+func (ew *EventWatcher) Watch(ctx context.Context) (<-chan []*models.KeptnContextExtendedCE, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(ctx)
+	ch := make(chan []*models.KeptnContextExtendedCE)
+	go ew.fetch(ctx, cancel, ch, ew.eventFilter)
+	return ch, cancel
+}
```
Comment (christian-kreuzberger-dtx) [inline pkg/api/utils/eventwatch.go:28]: This sounds like an important utility function for devs :+1:

Can you please add a usage example in our [README#usage](https://github.com/keptn/go-utils#usage)?

Comment (warber) [reply pkg/api/utils/eventwatch.go:28]: Done, I've also took the opportunity to cleanup the README file in general.

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM
thanks for the updates in readme!

**Pull request merged**